website/blog/2023-04-19-running-a-local-kubernetes-cluster-with-podman-desktop.md

```diff
@@ -0,0 +1,272 @@
+
+The guestbook app uses a PHP frontend.
+It is configured to communicate with either the Redis follower or leader Services, depending on whether the request is a read or a write.
+The frontend exposes a JSON interface, and serves a jQuery-Ajax-based UX.
+
+With Podman Desktop, you can prepare the Guestbook frontend image and container on your local container engine, and deploy the results to Kubernetes pods and services.
+This is functionally equal to the `frontend` deployment that the Kubernetes example propose.
+
+#### Procedure
+
+1. Open **<icon icon="fa-solid fa-cloud" size="lg" /> Images > <icon icon="fa-solid fa-arrow-circle-down" size="lg" /> Pull an image**.
+   1. **Image to Pull**: type `gcr.io/google_samples/gb-frontend:v5`
+   1. Click **Pull image** to pull the image to your container engine local image registry.
+   1. Wait for the pull to complete.
+   1. Click **Done** to get back to the images list.
+1. **<icon icon="fa-solid fa-cloud" size="lg" /> Search images**: type `gb-frontend:v5` to find the image.
+
+1. Click **<icon icon="fa-solid fa-ellipsis-v" size="lg" /> > <icon icon="fa-solid fa-ellipsis-v" size="lg" /> Push image to Kind cluster**.
+1. Click **<icon icon="fa-solid fa-play" size="lg" />** to open the **Create a container from image** dialog.
+   1. **Container name**: type `frontend`,
+   1. **Local port for `80/tcp`**: `9000`.
+   1. Click **<icon icon="fa-solid fa-play" size="lg" /> Start Container** to start the container in your container engine.
+1. **<icon icon="fa-solid fa-cloud" size="lg" /> Search containers**: type `frontend` to find the running container.
+1. Click **<icon icon="fa-solid fa-stop" size="lg" />** to stop the container: you don't need it to run in the container engine.
+1. Click **<icon icon="fa-solid fa-ellipsis-v" size="lg" /> > <icon icon="fa-solid fa-rocket" size="lg" /> Deploy to Kubernetes** to open the **Deploy generated pod to Kubernetes** screen.
+   1. **Pod Name**: type `frontend`.
+   1. **Use Kubernetes Services**: select **Replace .hostPort exposure on containers by Services. It is the recommended way to expose ports, as a cluster policy might prevent to use hostPort.**
+   1. **Kubernetes namespaces**: select `default`.
+   1. Click **<icon icon="fa-solid fa-rocket" size="lg" /> Deploy**.
+   1. Wait for the pod to reach the state: **Phase: Running**.
+   1. Click **Done**.
+
+#### Verification
+
+1. Open **<icon icon="fa-solid fa-cubes" size="lg" /> Pods**
+1. The pods list has a running `frontend` pod.
+
+## Exposing the Guestbook frontend service
+
+Podman Desktop installs Kind with the Contour Ingress controller.
+Use Ingress to expose the Guestbook frontend.
+
+#### Procedure
+
+1. Create a `frontend-ingress.yaml` file:
+
+   ```yaml
+   apiVersion: networking.k8s.io/v1
+   kind: Ingress
+   metadata:
+     name: frontend-ingress
+     annotations:
+       nginx.ingress.kubernetes.io/rewrite-target: /
```

Comment:
contour is not the nginx ingress controller, have to remove this.

Reply:
Oh,what is our ingress controller then?

I guess we have to fix the **Create a Kind cluster page** that states:

> Setup an ingress controller (Contour https://projectcontour.io):

Reply:
Oh no, I misunderstood, but now I get it: this YAML is for the nginx Ingress controller. We need to adapt the YAML for Contour.